app/controllers/order_controller.rb

```diff
@@ -0,0 +1,133 @@
+class OrderController < ApplicationController
+  before_action :authenticate_user!
+  before_action :current_user_cart
+
+  def index
+    @orders = current_user_orders.order(created_at: :desc)
+  end
+
+  def show
+    @order = current_user_orders.find_by(id: params[:id])
+  end
+
+  def checkout
+    @books = @cart.books
+    @total_price = @books.sum(&:price)
+  end
+
+  def payment
+    begin
+      ActiveRecord::Base.transaction do
+        books = @cart.books
+        total_price = books.sum(&:price)
+        CartItem.where(cart_id: @cart.id).delete_all
+
+        @order = current_user.orders.new(
+          total_price: total_price,
+          status: :pending,
+          shipping_address: order_params[:shipping_address]
+        )
+
+        books.each do |book|
+          @order.order_books.new(
+            book: book,
+            book_title: book.title,
+            price_per_unit: book.price,
+            quantity: 1
+          )
+        end
+
+        @order.save!
+
+        payment_intent = Stripe::PaymentIntent.create(
+          amount: (total_price * 100).to_i,
+          currency: Constants::CURRENCY,
+          payment_method_types: Constants::STRIPE_PAYMENT_METHODS,
+          metadata: {
+            order_id: @order.id,
+            user_id: current_user.id,
+            book_ids: books.ids.join(",")
+          }
+        )
```

Comment:
Should this be outside of the transaction? Because if there are any errors in line 55, the whole transaction is rolled back, but payment intent is already created in Stripe.

Reply:
yup, these codes here should be outside of the transaction.